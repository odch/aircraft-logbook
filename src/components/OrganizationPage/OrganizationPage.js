import React from 'react'
import PropTypes from 'prop-types'
import Header from '../../containers/HeaderContainer'
import OrganizationDetail from '../../containers/OrganizationDetailContainer'

const OrganizationPage = ({ match }) => (
  <React.Fragment>
    <Header />
    <OrganizationDetail match={match} />
  </React.Fragment>
)

OrganizationPage.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.object.isRequired
  }).isRequired
}

export default OrganizationPage
