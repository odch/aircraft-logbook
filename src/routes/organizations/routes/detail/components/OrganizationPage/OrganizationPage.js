import React from 'react'
import PropTypes from 'prop-types'
import Header from '../../../../../../containers/HeaderContainer'
import OrganizationDetail from '../../containers/OrganizationDetailContainer'

const OrganizationPage = ({ router: { match } }) => (
  <React.Fragment>
    <Header />
    <OrganizationDetail match={match} />
  </React.Fragment>
)

OrganizationPage.propTypes = {
  router: PropTypes.shape({
    match: PropTypes.shape({
      params: PropTypes.object.isRequired
    }).isRequired
  })
}

export default OrganizationPage
