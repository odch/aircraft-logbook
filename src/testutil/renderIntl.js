import React from 'react'
import renderer from 'react-test-renderer'
import { createMount } from '@material-ui/core/test-utils'
import { IntlProvider } from 'react-intl'
import pretty from 'pretty'
import messages from '../messages'

const LOCALE = 'de'

const renderInIntlProvider = component => (
  <IntlProvider locale={LOCALE} messages={messages[LOCALE]}>
    {component}
  </IntlProvider>
)

const renderIntl = (component, options) =>
  renderer.create(renderInIntlProvider(component), options)

export const renderIntlUpdate = (renderer, component, options) =>
  renderer.update(renderInIntlProvider(component), options)

export const renderIntlMaterial = (component, returnPrettyHtml = false) => {
  const rendered = createMount()(renderInIntlProvider(component))
  return returnPrettyHtml ? pretty(rendered.html()) : rendered
}

export default renderIntl
