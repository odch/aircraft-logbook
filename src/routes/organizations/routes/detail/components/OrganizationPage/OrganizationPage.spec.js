import React from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter as Router } from 'react-router-dom'
import configureStore from 'redux-mock-store'
import renderIntl from '../../../../../../testutil/renderIntl'
import OrganizationPage from './OrganizationPage'

describe('routes', () => {
  describe('organizations', () => {
    describe('routes', () => {
      describe('detail', () => {
        describe('components', () => {
          describe('OrganizationPage', () => {
            it('renders correctly', () => {
              const state = {
                firebase: {
                  auth: {
                    isEmpty: false
                  },
                  profile: {}
                },
                main: {
                  app: {
                    organizations: [{ id: 'my_org' }]
                  }
                },
                firestore: {
                  ordered: {
                    organizationAircrafts: [
                      {
                        id: 'o7flC7jw8jmkOfWo8oyA',
                        registration: 'HBKFW'
                      },
                      {
                        id: 'BKi7HYAIoe1i75H3LMk1',
                        registration: 'HBKOF'
                      }
                    ]
                  }
                }
              }

              const store = configureStore()(state)

              const props = {
                router: {
                  match: {
                    params: {
                      organizationId: 'my_org'
                    }
                  }
                }
              }

              const tree = renderIntl(
                <Provider store={store}>
                  <Router>
                    <OrganizationPage {...props} />
                  </Router>
                </Provider>
              ).toJSON()
              expect(tree).toMatchSnapshot()
            })
          })
        })
      })
    })
  })
})
