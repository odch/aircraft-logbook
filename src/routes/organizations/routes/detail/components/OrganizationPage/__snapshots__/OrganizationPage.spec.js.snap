// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes organizations routes detail components OrganizationPage renders correctly 1`] = `
Array [
  <header
    className="MuiPaper-root-13 MuiPaper-elevation4-19 MuiAppBar-root-4 MuiAppBar-positionStatic-8 MuiAppBar-colorPrimary-11"
  >
    <div
      className="MuiToolbar-root-40 MuiToolbar-regular-42 MuiToolbar-gutters-41"
    >
      <button
        aria-label="Menu"
        className="MuiButtonBase-root-50 MuiIconButton-root-44 MuiIconButton-colorInherit-45 Header-menuButton-3"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-49"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-53"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-62"
        />
      </button>
      <h2
        className="MuiTypography-root-69 MuiTypography-title-75 MuiTypography-colorInherit-89 Header-grow-2"
      >
        Open Digital Flugreisebuch
      </h2>
      <div>
        <button
          aria-haspopup="true"
          aria-owns={null}
          className="MuiButtonBase-root-50 MuiIconButton-root-44 MuiIconButton-colorInherit-45"
          data-cy="user-button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-49"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-53"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-62"
          />
        </button>
      </div>
    </div>
  </header>,
  <div
    className="OrganizationDetail-container-99"
  >
    <h2
      className="MuiTypography-root-69 MuiTypography-title-75"
    >
      my_org
    </h2>
    <div
      className="MuiGrid-container-104 MuiGrid-spacing-xs-24-128 AircraftList-container-100"
      data-cy="aircraft-list"
    >
      <div
        className="MuiGrid-item-105 MuiGrid-grid-xs-12-144 MuiGrid-grid-sm-4-150"
      >
        <div
          className="MuiPaper-root-13 MuiPaper-elevation1-16 MuiPaper-rounded-14 MuiCard-root-201 AircraftList-card-101"
        >
          <a
            className="MuiButtonBase-root-50 MuiCardActionArea-root-202 AircraftList-actionArea-103"
            href="/organizations/my_org/aircrafts/o7flC7jw8jmkOfWo8oyA"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
          >
            <div
              className="MuiCardContent-root-205 AircraftList-cardContent-102"
            >
              <h2
                className="MuiTypography-root-69 MuiTypography-title-75"
              >
                HBKFW
              </h2>
            </div>
            <span
              className="MuiCardActionArea-focusHighlight-204"
            />
            <span
              className="MuiTouchRipple-root-62"
            />
          </a>
        </div>
      </div>
      <div
        className="MuiGrid-item-105 MuiGrid-grid-xs-12-144 MuiGrid-grid-sm-4-150"
      >
        <div
          className="MuiPaper-root-13 MuiPaper-elevation1-16 MuiPaper-rounded-14 MuiCard-root-201 AircraftList-card-101"
        >
          <a
            className="MuiButtonBase-root-50 MuiCardActionArea-root-202 AircraftList-actionArea-103"
            href="/organizations/my_org/aircrafts/BKi7HYAIoe1i75H3LMk1"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
          >
            <div
              className="MuiCardContent-root-205 AircraftList-cardContent-102"
            >
              <h2
                className="MuiTypography-root-69 MuiTypography-title-75"
              >
                HBKOF
              </h2>
            </div>
            <span
              className="MuiCardActionArea-focusHighlight-204"
            />
            <span
              className="MuiTouchRipple-root-62"
            />
          </a>
        </div>
      </div>
    </div>
  </div>,
]
`;
